connector/src/main/java/org/geysermc/connector/utils/SettingsUtils.java

```diff
@@ -121,13 +134,23 @@ public static boolean handleSettingsForm(GeyserSession session, String response)
         }
         int offset = 0;
 
-        // Client can only see its coordinates if reducedDebugInfo is disabled and coordinates are enabled in geyser config.
-        if (!session.isReducedDebugInfo() && session.getConnector().getConfig().isShowCoordinates()) {
+        if (session.getPreferencesCache().isAllowShowCoordinates() || !(CooldownUtils.getDefaultShowCooldown() == CooldownUtils.CooldownType.DISABLED)) {
             offset++; // Client settings title
 
-            session.getWorldCache().setPrefersShowCoordinates(settingsResponse.getToggleResponses().get(offset));
-            session.getWorldCache().updateShowCoordinates();
-            offset++;
+            // Client can only see its coordinates if reducedDebugInfo is disabled and coordinates are enabled in geyser config.
+            if (session.getPreferencesCache().isAllowShowCoordinates()) {
+                session.getPreferencesCache().setPrefersShowCoordinates(settingsResponse.getToggleResponses().get(offset));
+                session.getPreferencesCache().updateShowCoordinates();
+                offset++;
+            }
+
+            if (!(CooldownUtils.getDefaultShowCooldown() == CooldownUtils.CooldownType.DISABLED)) {
```

Comment:
Same as the last comment.

Reply:
addressed in 8767aacf545324b095e599bb307afe51ac7d0ea1